test/audiovideocontroller/DefaultAudioVideoController.test.ts

```diff
@@ -1539,17 +1529,14 @@ describe('DefaultAudioVideoController', () => {
       audioVideoController.meetingSessionContext.videoStreamIndex = new TestVideoStreamIndex();
       // @ts-ignore
       audioVideoController.meetingSessionContext.transceiverController = new TestTransceiverController();
-      // @ts-ignore
-      audioVideoController.meetingSessionContext.signalingClient = new TestSignalingClient(
-        webSocketAdapter,
-        new NoOpDebugLogger()
-      );
 
+      // One that requires subscribe
+      audioVideoController.update({ needsRenegotiation: false });
+      // One that doesn't
```

Comment:
this code confuses me :)

Reply:
The unit testing war against async code (and components with many dependencies) will likely never be won. But this unit test was originally confirming the bug in the code :P ...

Reply:
So the confusing part is `needsRenegotiation` is both set to false but the comment says one requires subcribe and one doesn't.  :D

Reply:
`needsRenegotiation` could also have been named `forceRenegotiation` :)